pkg/resources/dynamic_table_acceptance_test.go

```diff
@@ -0,0 +1,118 @@
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: testAccCheckDynamicTableDestroy,
+		Steps: []resource.TestStep{
+			{
+				ConfigDirectory: config.TestStepDirectory(),
+				ConfigVariables: variableSet1,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttr(resourceName, "database", acc.TestDatabaseName),
+					resource.TestCheckResourceAttr(resourceName, "schema", acc.TestSchemaName),
+					resource.TestCheckResourceAttr(resourceName, "warehouse", acc.TestWarehouseName),
+					resource.TestCheckResourceAttr(resourceName, "target_lag.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "target_lag.0.maximum_duration", "2 minutes"),
+					resource.TestCheckResourceAttr(resourceName, "query", fmt.Sprintf("select \"id\" from \"%v\".\"%v\".\"%v\"", acc.TestDatabaseName, acc.TestSchemaName, tableName)),
+					resource.TestCheckResourceAttr(resourceName, "comment", "Terraform acceptance test"),
+
+					// computed attributes
+
+					// - not used at this time
+					//  resource.TestCheckResourceAttrSet(resourceName, "cluster_by"),
+					resource.TestCheckResourceAttrSet(resourceName, "rows"),
+					resource.TestCheckResourceAttrSet(resourceName, "bytes"),
+					resource.TestCheckResourceAttrSet(resourceName, "owner"),
+					resource.TestCheckResourceAttrSet(resourceName, "refresh_mode"),
+					// - not used at this time
+					// resource.TestCheckResourceAttrSet(resourceName, "automatic_clustering"),
+					resource.TestCheckResourceAttrSet(resourceName, "scheduling_state"),
+					resource.TestCheckResourceAttrSet(resourceName, "last_suspended_on"),
+					resource.TestCheckResourceAttrSet(resourceName, "is_clone"),
+					resource.TestCheckResourceAttrSet(resourceName, "is_replica"),
+					resource.TestCheckResourceAttrSet(resourceName, "data_timestamp"),
+				),
+			},
+			// test target lag to downstream and change comment
+
+			{
+				ConfigDirectory: config.TestStepDirectory(),
+				ConfigVariables: variableSet2,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttr(resourceName, "database", acc.TestDatabaseName),
+					resource.TestCheckResourceAttr(resourceName, "schema", acc.TestSchemaName),
+					resource.TestCheckResourceAttr(resourceName, "target_lag.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "target_lag.0.downstream", "true"),
+					resource.TestCheckResourceAttr(resourceName, "comment", "Terraform acceptance test - updated"),
+				),
+			},
+			// test import
+			{
+				ConfigDirectory:   config.TestStepDirectory(),
+				ConfigVariables:   variableSet2,
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckDynamicTableDestroy(s *terraform.State) error {
```

Comment:
I think check destroy functions could be made into one common function that would do the same logic 

and for uncommon things like 
```go
id := sdk.NewSchemaObjectIdentifier(rs.Primary.Attributes["database"], rs.Primary.Attributes["schema"], rs.Primary.Attributes["name"])
dynamicTable, err := client.DynamicTables.ShowByID(ctx, id)
```
there would be a function passed as a parameter that receives resource ("rs") and client and returns an error that is common operation (or the err could be checked in the function)

Reply:
true, but i don't want to do that right now. i think we will need to have a separate discussion on this, similar to what was done for integration tests. for now, at least it works